Mathlib/Algebra/Polynomial/Roots.lean

```diff
@@ -609,9 +609,21 @@ lemma exists_eval_ne_zero_of_natDegree_lt_card (f : R[X]) (hf : f ≠ 0) (hfR :
   contrapose! hf
   exact eq_zero_of_forall_eval_zero_of_natDegree_lt_card f hf hfR
 
-theorem monic_prod_multiset_X_sub_C : Monic (p.roots.map fun a => X - C a).prod :=
+section
+
+omit [IsDomain R]
+
+theorem monic_prod_multiset_X_sub_C (s : Multiset R) : Monic (s.map fun a => X - C a).prod :=
   monic_multiset_prod_of_monic _ _ fun a _ => monic_X_sub_C a
 
+theorem monic_finset_prod_X_sub_C (s : Finset R) : Monic (∏ a ∈ s, (X - C a)) :=
```

Comment:
```suggestion
theorem monic_finset_prod_X_sub_C {α : Type*} (b : α → R) (s : Finset α) : Monic (∏ i ∈ s, (X - C (b i))) :=
```
Hopefully the proof just works

Reply:
And should we omit `finset` from the name? It seems `Finset.prod` is the default product and [Polynomial.monic_prod_of_monic](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Monic.html#Polynomial.monic_prod_of_monic), [map_prod](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/BigOperators/Group/Finset/Defs.html#map_prod) etc. don't have Finset in its name.

Reply:
And arguably `monic_prod_multiset_X_sub_C` should be `monic_multiset_prod_X_sub_C` instead...
It's fine if it is not done in this PR.

Reply:
`multiProd`, rather. There are [some decls](https://loogle.lean-lang.org/?q=%22multisetprod%22) that already adopted this convention.